nemo_curator/scripts/semdedup/compute_embeddings.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import time
+from typing import List
+
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.log import create_logger
+from nemo_curator.modules.config import SemDedupConfig
+from nemo_curator.modules.semantic_dedup import EmbeddingCreator
+from nemo_curator.utils.distributed_utils import get_client, read_data
+from nemo_curator.utils.file_utils import get_remaining_files
+from nemo_curator.utils.script_utils import ArgumentHelper
+
+
+def get_input_files(
```

Comment:
Couple of things here:
- After reading the body of this function, the `num_samples` parameter is misleading in its name. A sample typically refers to a single document, while in this case it appears to be referring to a file. Can this be renamed to `num_files` or `num_shards`?
- I am not a fan of having another unique function for reading files/figuring out what files to be read. It makes code much more confusing and harder to maintain. I want to enforce some kind of consistency. Even within semantic dedup, each of the three CLI scripts have a different way of reading in files:
  - `compute_embeddings.py` (this script) uses `read_data` with the new `get_input_files` function.
  - `clustering.py` uses `dask_cudf.read_parquet`.
  - `extract_dedup_data.py` reads the files in deep in `SemanticClusterLevelDedup.compute_semantic_match_dfs`, which eventually calls `cudf.read_parquet`
   
  We already have [noted](https://github.com/NVIDIA/NeMo-Curator/issues/50) that our file operations are not easy to work with, and our [future plans](https://github.com/NVIDIA/NeMo-Curator/issues/66) are only going to get harder as we introduce more ways of reading in files.
   
  The way I want users (and us) to read in files (right now) is this:
    - Use `DocumentDataset.read_*` whenever you know the datatype at the time of writing the script.
    - Use `read_data` whenever you don't. We should eventually make a similar function directly in `DocumentDataset`, but that's beside the point.
 
  `read_data` should be the way to go with in the CLI scripts. `get_remaining_files` or `get_all_files_paths_under` can be helpers for that function if needed (I'm not a fan of having two helpers in the first place either, but again, beside the point). I'd rather not have a new helper method like this in the mix too. In this case, perhaps we could merge this function with the `get_remaining_files` function. See my comment below for more on that.

  Furthermore, we shouldn't need to be working around our file operations. If we feel that we need to do that, we should modify them instead to fit our usecase. I know we're on a crunch right now, but anything you can do to get us closer to the ideal case I mentioned above would be great.

Reply:
I agree with the spirit of having consistent IO format but we wont be able to do it till we address  https://github.com/NVIDIA/NeMo-Curator/issues/50 , like 

- compute_embeddings.py (this script) uses read_data with the new get_input_files function.
Agreed, merging with `read_data`. 

- clustering.py uses dask_cudf.read_parquet because we don't have  a block-wise support which is important from performance. Once we fix it , i am happy to revisit this. 

- SemanticClusterLevelDedup.compute_semantic_match_dfs calls cudf.read_parquet, Unfortunately there is no straightforward way for this .  We should pick the tool as needed especially for complex workflows so i think we are stuck there.

For now, I will link  https://github.com/NVIDIA/NeMo-Curator/issues/50  here and merge get_remaining_files . I hope thats a good middle path.

Reply:
Yeah sounds good to me. Wasn't trying to put this all on you right now, just making sure I get all my thoughts put down.

Reply:
Done